### PR TITLE
Fix typo & minor edits to security NVD doc

### PR DESCRIPTION
**Pull request by michelle-purcell**
This PR fixes a doc typo and an obsolete link label.

Plus enhance and apply style as per the [Quarkus docs contributor style guide](https://quarkus.io/guides/doc-reference).

### CODE REVIEW

**Comment by michelle-purcell** [comment]
@sberyozkin - Another small one for your review :eyes: Thank you!

File: docs/src/main/asciidoc/security-vulnerability-detection.adoc [hunk 1]
```diff
@@ -130,9 +130,10 @@ Ensure that you review and update the suppression list regularly to ensure that
...
 
 `<suppress until="2022-01-01Z">...</suppress>`
+
 You can adjust the expiry date if you need to.
 
 == References
 
 * xref:security-overview.adoc[Quarkus Security overview]
-* xref:security-authentication-mechanisms.adoc#other-supported-authentication-mechanisms[Authentication mechanisms in Quarkus]
+* xref:security-authentication-mechanisms.adoc#other-supported-authentication-mechanisms[Other supported authentication mechanisms]
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-vulnerability-detection.adoc:139]:
```suggestion
* xref:security-authentication-mechanisms.adoc[Supported authentication mechanisms]
```

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-vulnerability-detection.adoc:139]: And the doc title is `Authentication mechanisms in Quarkus` so I just updated the suggestion

Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-vulnerability-detection.adoc]: Fixed.

**Review by sberyozkin — APPROVED** [review]
Thanks Michelle

**Pull request merged**